sdks/go/pkg/logger/logger.go

```
@@ -38,118 +38,216 @@
 package logger
 
 import (
+	"context"
 	"fmt"
 	"io"
+	"log/slog"
 	"os"
 	"path/filepath"
 	"sync"
 	"time"
 )
 
 var (
-    globalLogger *Logger
-    once         sync.Once
+	globalLogger *Logger
+	once         sync.Once
 )
 
-// Logger handles logging to console and file
+// Logger handles structured logging using slog
 type Logger struct {
-    mu         sync.Mutex
-    config     Config
-    file       *os.File
-    writers    []io.Writer
+	config  Config
+	slogger *slog.Logger
+	file    *os.File
 }
 
-// Init initializes the global logger (matches C++ Logger::init)
+// Init initializes the global logger
 func Init(cfg Config) error {
```

should probably throw an error or something if this is called multiple times to indicate to the user that they can't re-init logging